network/src/node.rs

```diff
@@ -229,13 +231,39 @@ impl Node {
         self.register_task(state_tracking_task);
 
         if self.sync().is_some() {
-            let bootnodes = self.config.bootnodes();
+            let node_clone = self.clone();
+            let block_sync_interval = node_clone.expect_sync().block_sync_interval();
+            let sync_block_task = task::spawn(async move {
+                loop {
+                    let is_syncing_blocks = node_clone.is_syncing_blocks();
+
+                    if !is_syncing_blocks {
+                        if let Err(e) = node_clone.run_sync().await {
+                            error!("failed sync process: {:?}", e);
+                        }
+                        node_clone.finished_syncing_blocks();
+                    }
+
+                    sleep(block_sync_interval).await;
+                }
+            });
+            self.register_task(sync_block_task);
 
+            let bootnodes = self.config.bootnodes();
             let node_clone = self.clone();
             let mempool_sync_interval = node_clone.expect_sync().mempool_sync_interval();
             let sync_mempool_task = task::spawn(async move {
                 loop {
                     if !node_clone.is_syncing_blocks() {
+                        // Ensure that it's not just a short "break" from syncing blocks
+                        #[cfg(not(feature = "test"))]
+                        if let Some(elapsed) = node_clone.expect_sync().time_since_last_block_sync() {
+                            if elapsed < Duration::from_secs(60) {
```

Comment:
What was there a particular reason for choosing `60s`?

Reply:
Yes; I initially tried to make it dependent on the block sync interval, but I quickly needed to multiply it by a relatively high value in order for the delay to be sufficient; I later observed how long a single sync batch could be processed in a few test runs (it was ~30s for some problematic cases) and just decided to add an additional safety margin for slower PCs.